core/templates/dev/head/domain/exploration/AnswerGroupObjectFactorySpecs.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview unit tests for answer group object factory.
+ */
+
+import { TestBed } from '@angular/core/testing';
+
+import { AnswerGroupObjectFactory } from
+  'domain/exploration/AnswerGroupObjectFactory';
+import { OutcomeObjectFactory } from 'domain/exploration/OutcomeObjectFactory';
+
+describe('Answer Group Object Factory', () => {
+  let agof: AnswerGroupObjectFactory;
+  let oof: OutcomeObjectFactory;
+  let testAnswerGroup = agof.createNew([], oof.createNew('dest_1',
```

Comment:
1. is agof and oof defined before?
2. If testAnswerGroup is used in different test cases then I would suggest defining this in the beforeEach section so that all the test blocks are independent.

Reply:
1)they are defined in the before each section.
2) Done

Reply:
While definining them in the before each section I am getting errors.

Reply:
Made the changes

Reply:
they are defined in the before each section